dev/tasks/r/github.nightly.yml

```diff
@@ -0,0 +1,467 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+{% import 'macros.jinja' as macros with context %}
+# root of the R nightly repo where the first dir is the mount point used by davfs2 ("nightly")
+# TODO update with nightlies.apache.org/arrow/r
+{% set r_root = "nightly/r" %}
+{% set webdav_domain = "https://nightly.wujciak.de:8080" %}
+{% set repo_domain = "https://nightly.wujciak.de/r" %}
+# This allows us to set a custom version via param:
+# crossbow submit --param custom_version=8.5.3 r-nightly-packages
+# if the param is unset defaults to the usual Ymd naming scheme
+{% set version = custom_version|default('\\2.\'\"$(date +%Y%m%d)\"\'') %}
+# We need this as boolean and string
+{% set is_upstream_b = arrow.github_repo == 'apache/arrow' %}
+# use filter to cast to string and convert to lowercase to match yaml boolean
+{% set is_fork = (not is_upstream_b)|lower %}
+{% set is_upstream = is_upstream_b|lower %}
+{% set upload_pkg = "env.UPLOAD_PKG == 'true' && success() && " + is_upstream %}
+{% set upload_libarrow = "env.UPLOAD_LIBARROW == 'true' && success() && " + is_upstream %}

```

Comment:
This is a pretty heavy workflow :)

Until recently we preferred smaller crossbow tasks so we can trigger them separately.
Of course it's not an issue if the tasks are too dependent on each other.

Reply:
Yes, it was great fun to test and debug 😅 Usually I would agree to divide it into smaller workflows but the jobs are dependent on each other and the workflow's main purpose is to produce the nightlies, while it can be run on PRs I think this should be an exception e.g. during release/pre-cran.